agent/api/task/task.go

```diff
@@ -1773,10 +1826,18 @@ func (task *Task) shouldOverrideNetworkMode(container *apicontainer.Container, d
 	// TODO. We can do an early return here by determining which kind of task it is
 	// Example: Does this task have ENIs in its payload, what is its networking mode etc
 	if container.IsInternal() {
-		// If it's an internal container, set the network mode to none.
+		// If it's an internal container, set the network mode to none for awsvpc, set to bridge if task is using
+		// bridge mode and this is an SC-enabled task.
 		// Currently, internal containers are either for creating empty host
 		// volumes or for creating the 'pause' container. Both of these
 		// only need the network mode to be set to "none"
+		if container.Type == apicontainer.ContainerCNIPause {
+			if task.IsNetworkModeAWSVPC() {
+				return true, networkModeNone
+			} else if task.IsNetworkModeBridge() && task.IsServiceConnectEnabled() {
+				return true, BridgeNetworkMode
```

Comment:
Why are we not setting it to networkModeNone as well?

Reply:
Because for bridge mode, the pause containers have to start in bridge mode, so that when the real customer containers have bridge network connectivity when they join its respective pause container

Reply:
Next PR (by Yinyi) will make this more clear. But this is by design.

Reply:
got it.